ceph_salt/core.py

```diff
@@ -39,7 +39,15 @@ def _load(self):
             self.execution = result[self.minion_id]['execution']
 
         result = GrainsManager.get_grain(self.minion_id, 'fqdn_ip4')
-        self.public_ip = result[self.minion_id][0]
+        public_ip = result[self.minion_id][0]
+        if public_ip == '127.0.0.1':
+            logger.debug('fqdn_ipv4 grain is 127.0.0.1, falling back to ipv4 grain')
+            result = GrainsManager.get_grain(self.minion_id, 'ipv4')
+            for addr in result[self.minion_id]:
+                if addr != '127.0.0.1':
+                    public_ip = addr
+                    break
+        self.public_ip = public_ip
```

Comment:
Shouldn't that fail here if we still have `127.0.0.1`?

Reply:
It will fail latter when you run `ceph-salt status` or `ceph-salt deploy` - https://github.com/ceph/ceph-salt/blob/master/ceph_salt/validate/config.py#L20

Reply:
This scenario (`public_ip` still coming up as `127.0.0.1`) would mean that either the node has no ip besides `127.0.0.1` or that salt is messing up populating the ipv4 grain.

We can probably ignore the first case, since we would not have a functioning salt cluster. For the second case at least a log message would be nice I guess.

Reply:
https://github.com/ceph/ceph-salt/issues/183